src/dsp_tools/utils/uri_util.py

```diff
@@ -12,3 +12,55 @@ def is_uri(s: str) -> bool:
     fragment = r"(?<fragment>#[\w_.\-~:/]*)"
     m = regex.match(rf"{scheme}:(//{host}{port}?){path}*{query}*{fragment}?", s, flags=regex.UNICODE)
     return m is not None
+
+
+def is_iiif_uri(uri: str) -> bool:
+    """
+    Checks if the given URL is a valid IIIF URL.
+    It should support all versions of IIIF servers,
+    but was constructed with the syntax described in: https://iiif.io/api/image/3.0/#2-uri-syntax
+    IIIF is open to modifications which are not described in the official documentation.
+    https://iiif.io/api/annex/notes/design_principles/#define-success-not-failure
+    This may need to be updated in case of a valid version which is not in the regex.
+
+    Args:
+        uri: The URI to be checked.
+
+    Returns:
+        True if the URI is a valid IIIF URI, False otherwise.
+    """
+    # {scheme}://{server}{/prefix}/{identifier}/{region}/{size}/{rotation}/{quality}.{format}
+    split_uri = uri.lower().split("/")
+    if len(split_uri) < 8:
```

Comment:
If the example above is splitted by `/`, I count 9 parts, not 8

Reply:
Everything before region is not regulated strictly, so this is just an approximation. This is the general problem with this regex, they have a specification, but also state that they are open for other implementations.